src/lib.rs

```diff
@@ -283,10 +274,20 @@ impl TryFrom<Vec<u8>> for PrivateKeyDer<'_> {
     type Error = &'static str;
 
     fn try_from(key: Vec<u8>) -> Result<Self, Self::Error> {
-        Ok(match PrivateKeyDer::try_from(&key[..])? {
-            PrivateKeyDer::Pkcs1(_) => Self::Pkcs1(key.into()),
-            PrivateKeyDer::Sec1(_) => Self::Sec1(key.into()),
-            PrivateKeyDer::Pkcs8(_) => Self::Pkcs8(key.into()),
+        let key_ref = PrivateKeyDer::try_from(&key[..])?;
+        Ok(match &key_ref {
+            PrivateKeyDer::Pkcs1(_) => {
+                drop(key_ref);
```

Comment:
This slightly awkward restructuring is to avoid bothering the borrow checker, which otherwise complains that the borrowing `PrivateKeyDer` value needs to run `Drop` at the end of the function, after moving `key` into the owning `PrivateKeyDer`.

Reply:
Is this sort of thing likely to happen also in calling code? (I can't really find any clear declaration that impling Drop is a minor change, but I do have a bit set that it varies the borrow checker behaviour in some circumstances -- [this perhaps?](https://doc.rust-lang.org/std/ops/trait.Drop.html#drop-check) -- but it isn't mentioned at all in the semver rules.)

Reply:
I think it could happen, but would probably be unlikely? I'm having a hard time coming with up with a specific example.

Given the lack of specific mention of `impl Drop` in the semver guidelines, I guess it should be treated like any other trait impl, and thus be classified as minor? I filed [an issue](https://github.com/rust-lang/cargo/issues/15471) against the SemVer reference guide.